From 1be69105abccec526afdf7b94611473312a931db Mon Sep 17 00:00:00 2001
From: Thierry FOURNIER <tfournier@exceliance.fr>
Date: Tue, 15 Apr 2014 01:38:48 +0200
Subject: [PATCH 138/139] BUG/MINOR: log: Don't dump empty unique-id

If the unique-id value is missing, the build_logline() function dump
anything. It is because the function lf_text() is bypassed. This
function is responsible to dump '-' is the value is not present, and set
the '"' around the value displayed.

This fixes the bug reported by Julient Vehent
---
 src/log.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/log.c b/src/log.c
index 176a725..efb3d27 100644
--- a/src/log.c
+++ b/src/log.c
@@ -1559,8 +1559,7 @@ int build_logline(struct session *s, char *dst, size_t maxsize, struct list *lis
 			case LOG_FMT_UNIQUEID: // %ID
 				ret = NULL;
 				src = s->unique_id;
-				if (src)
-					ret = lf_text(tmplog, src, maxsize - (tmplog - dst), tmp);
+				ret = lf_text(tmplog, src, maxsize - (tmplog - dst), tmp);
 				if (ret == NULL)
 					goto out;
 				tmplog = ret;
-- 
1.8.3.2

